Note component

Snezzz opened the pull request:
Created Note component and NoteFull component

Snezzz committed:
Created Note component

Snezzz committed:
Created Note component and NoteFull component
Everything works!

Snezzz committed:
Добавила два режима работы с компонентой "Note":
1.Режим содания: задаются дефолтные значения полей
2.Режим редактирвования: задаются существующие данные и сразу же режим
редактирования

mascarpony commented on src/App.js:
```diff
@@ -1,25 +1,16 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import Note from "./Components/Note/note";
```
Перед мержем убери заметки из app компонента, они будут лежать в компонентах страниц

mascarpony replied:
И если правильно сделаешь index.js файлы, то можно будет импортить `import Note from './Components/Note'`

mascarpony commented on src/Components/FullNote/index.js:
```diff
@@ -0,0 +1,3 @@
+import 'fullnote.js'
```
Этот файл должен выглядеть так:
```
import FullNote from './fullnote';

export default FullNote;
```
И на этом все :) css файл надо импортить в сам fullnote.js
То же самое касается note.js

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,190 @@
+        }
+        switch (k){
+            case 1:
+                this.state.first="firstTrue";
```
Строковые значения не очень хорошо использовать, особенно если у тебя есть всегт два значения. Вместо них лучше взять `true` и `false`. Ну и для этого переменная, которая хранится в стейте, должна иметь более говорящее название

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,190 @@
+                break;
+
+        }
+        this.handleMouseOver = this.handleMouseOver.bind(this);
```
Без этого точно не работает?

mascarpony commented on src/Components/FullNote/fullnote.css:
```diff
@@ -0,0 +1,119 @@
+    background-color: #ececec;
+    margin: 0 0 0 0 !important;
+}
+.importance .firstTrue,.secondTrue,.thirdTrue{
```
Ты используешь три разных класса для того, чтобы задавать одни и те же свойства. Вместо этого ты можешь сделать один класс и присваивать его элементам, если переданное значение тру. В разметке это может выглядеть так:
```
 <button className = {this.state.first ? 'active' : 'inactive'} onMouseOver={this.handleMouseOver} onMouseUp={this.handleMouseUp}></button>
```

Snezzz committed:
1.Исправила import-ы
2.Изменила логику работы с лампочками (true/false)
3.Добавила возможность добавлять теги. Удалять пока нельзя. Если добавляешь
тег, который уже существует, ничего не происходит. Можно добавить сразу несколько тегов
4.Переместила изображения в папку "src/icons"

Snezzz committed:
Created FullNoteContainer

Snezzz committed:
Updated FullNoteContainer and FullNote logic

mascarpony commented on src/Containers/FullNoteContainer/FullNoteContainer.js:
```diff
@@ -0,0 +1,60 @@
+    //update Note (from FullNote)
+    onSave() {
+            //POST
+            fetch("http://localhost:8083/note/", {
```
По поводу урла: будет классно сделать папку Constants в src, сделать файл index.js и из него экпортить константу `URL='http://localhost:8083/'`. Тогда, если что-то поменяется, то будет достаточно сделать изменение в одном месте

mascarpony commented on src/Containers/FullNoteContainer/FullNoteContainer.js:
```diff
@@ -0,0 +1,60 @@
+    //delete Note
+    onDelete(id){
+        //DELETE
+        fetch("http://localhost:8083/note/"+id,
```
Эту строчку можно переписать через современный синтаксис: ``` `http://localhost:8083/note/${id}` ```

mascarpony commented on src/Containers/FullNoteContainer/FullNoteContainer.js:
```diff
@@ -0,0 +1,60 @@
+                    this.setState({Note: response});
+                })
+                .then((error) => {
+                    this.setState({false: true});
```
Можно сделать это в одной функции setState

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,194 @@
+        super(props);
+        this.state = {
+            Note: props.Note, //если новая - пусто, если существующаяя - передаем ее
+            onSave: props.onSave,
```
Тебе не нужно записывать в стейт абсолютно все, функции можно взять и из пропсов) В стейте должны быть только те значения, которые в процессе существования компонента будут изменены самим компонентом

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,194 @@
+                    </div>
+                    <div className="importance">
+                        <div>
+                            <button className = {this.state.first ? 'activeLamp' : 'inactiveLamp'} onClick={this.setImportance}></button>
```
Подумала о том, что с лампочками можно сделать немного более простую логику: хранить в стейте значение importance (от 0 до 3), и если, допустим, на первой лампочке this.state.importance >= 1, то зажигать ее

Snezzz committed:
Updated FullNote,
Added arrows to get next/previous note

mascarpony commented on src/Containers/FullNoteContainer/FullNoteContainer.js:
```diff
@@ -0,0 +1,60 @@
+    //delete Note
+    onDelete(id){
+        //DELETE
+        fetch(URL+"/note/${id}",
```
Неправильные кавычки) Тут нужны обратные, их можно найти там, где буква ё. С ними эта строчка должна выглядеть так: ``` `${URL}note/${id}` ```
И также можно переписать все строчки с урлами

mascarpony replied:
Это синтаксис шаблонных строк, достаточно крутая штука: https://learn.javascript.ru/es-string

mascarpony commented on src/Components/Note/note.js:
```diff
@@ -0,0 +1,67 @@
+    }
+    handleAdd(){
+        const result = this.state.result;
+        if(result){
```
У тебя тут есть штуки, которые не нужны в компоненте отдельной заметки. Оставь только рисование самой заметки, остальное не нужно. И, думаю, сюда по аналогии с FullNote через пропсы надо прокидывать функцию, которая будет вызываться при клике на заметку

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,199 @@
+            }
+        }
+        switch (j){
+            case '0':
```
Думаю, тут можно сделать проще) Что-нибудь типа 
```
                this.setState({
                    importance : +j + 1
                });
```
А еще вообще не понятно, что такое j

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,199 @@
+                    </div>
+                    <div className="action">
+                        <button className={this.state.cancel} onMouseOver={this.handleMOver}
+                                onMouseOut={this.handleMouseOut}
```
Зачем здесь mouseOut? Может, оставить только клик?

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,199 @@
+    render(){
+        let note = this.state.Note;
+        //если новая заметка, создаем поля, создаем поле для добавления тегов
+        if(Object.keys(note).length===0){
```
Лучше сделать это сразу в стейте. Точнее, сделать вне этого компонента (например, до него) константу с пустой заметкой и записать в стейт эту пустую заметку, если пришло пустое значение

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,199 @@
+        }
+        //если обновляем старую заметку, здесь только формируется список тегов
+        else{
+            this.state.isNew = false; //для отображения стрелочек перехода к другой заметке
```
Тебе не нужно записывать это в стейт. Можно в самой разметке использовать `this.state.id === 'new' ? <здесь ваша разметка> : null`

mascarpony commented on src/Components/FullNote/fullnote.js:
```diff
@@ -0,0 +1,199 @@
+        //если обновляем старую заметку, здесь только формируется список тегов
+        else{
+            this.state.isNew = false; //для отображения стрелочек перехода к другой заметке
+            this.state.tag = <ul>{this.state.Note.tags.map((tag) =>
```
Это тоже не нужно записывать в стейт. Можно сделать просто переменную внутри функции рендер и в нее записать теги

Snezzz committed:
Updated FullNote, FullNoteContainer
In note.js Note2 is just for testing

Snezzz committed:
1.обработчики событий -> :hover
2.удалила лишнее из note.js
3.импортила контейнер в App.js

Snezzz committed:
corrections to haven't conflicts

Snezzz committed:
Merge branches 'master' and 'noteComponent' of https://github.com/ladalada/Idea-FE into noteComponent

# Conflicts:
#	README.md
#	src/App.js

Snezzz committed:
Merge branch 'master' into noteComponent

Snezzz committed:
Изменила немного логику создания, изменения и удаления заметки

Snezzz committed:
Merge remote-tracking branch 'origin/noteComponent' into noteComponent

Snezzz committed:
Добавлена возможность добавлять теги, но только из существующих в БД
